web/src/dialog_widget.ts

```diff
@@ -156,30 +186,30 @@ export function launch(conf) {
 }
 
 export function submit_api_request(
-    request_method,
-    url,
+    request_method: AjaxRequest,
+    url: string,
     data = {},
     {
         failure_msg_html = $t_html({defaultMessage: "Failed"}),
         success_continuation,
         error_continuation,
-    } = {},
-) {
+    }: RequestOpts = {},
+): void {
     show_dialog_spinner();
     request_method({
         url,
         data,
-        success(response_data) {
+        success(response_data?: string) {
             close_modal();
             if (success_continuation !== undefined) {
-                success_continuation(response_data);
+                success_continuation?.(response_data);
             }
```

Comment:
I'd think the `?` here would not be required given that we just checked it's not undefined.

Reply:
Noted! I will take care of this in my future work, also pinging @Ddharmani3 please have a look at this comment.

Reply:
Definitely, I will take care of it in my future work.

Reply:
> Noted! I will take care of this in my future work, also pinging @Ddharmani3 please have a look at this comment.